src/common/duplication_common.cpp

```diff
@@ -52,12 +54,9 @@ namespace replication {
     return it->second;
 }
 
-/*extern*/ const char *get_current_cluster_name()
-{
-    static const char *cluster_name =
-        dsn_config_get_value_string("replication", "cluster_name", "", "name of this cluster");
-    dassert(strlen(cluster_name) != 0, "cluster_name is not set");
-    return cluster_name;
+/*extern*/ const char *get_current_cluster_name() {
+    dassert(strlen(FLAGS_cluster_name) != 0, "cluster_name is not set");
```

Comment:
```suggestion
    dassert(strlen(FLAGS_cluster_name) != 0, "cluster_name is not set");
```

Use flag validator instead

Reply:
We can't use flag validator, because cpp client will not set cluster name. so the validator will always fail for cpp client